salt/utils/network.py

```diff
@@ -1958,3 +1959,45 @@ def parse_host_port(host_port):
             raise ValueError('bad hostname: "{}"'.format(host))
 
     return host, port
+
+
+def _filter_by_networks(values, networks):
+    '''
+    Filter address list against a list of networks.
+    '''
+    filtered_addresses = []
+
+    for value in values:
+        address = ipaddress.ip_address(value)
+        for network in networks:
+            if address in network:
+                filtered_addresses.append(value)
+
+    return filtered_addresses
```

Comment:
All this can be replaced with

```python
return [i for n in networks for i in values if ipaddress.ip_address(i) in n]
```

You can put that in a`lambda` inside the main function (or a inner function)

Reply:
For me the loop is more readable than the list comprehension expression.

Reply:
I think a list comprehension like this is both readable and simpler:
```python
    return [ip for ip in ips
               for net in networks
               if ipaddress.ip_address(ip) in net]
```
If I don't overlook something the runtime of this is `len(ips) * len(networks)` in either case, so switching the loops (compared to @aplanas suggestion) does not matter.

Reply:
@agraul the complexity is the same, yes. Choose whatever you want : )